src/wallet/api/wallet_manager.cpp

```diff
@@ -222,38 +214,23 @@ bool WalletManagerImpl::checkPayment(const std::string &address_text, const std:
     return false;
   }
 
-  cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request req;
-  cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response res;
-  req.txs_hashes.push_back(epee::string_tools::pod_to_hex(txid));
-  if (!connect_and_invoke(m_daemonAddress, "/gettransactions", req, res) ||
-      (res.txs.size() != 1 && res.txs_as_hex.size() != 1))
+  std::vector<crypto::hash> tx_hashes;
+  tx_hashes.push_back(txid);
+
+  std::unordered_map<crypto::hash, cryptonote::rpc::transaction_info> txs;
+  std::vector<crypto::hash> missed_hashes;
+
+  boost::optional<std::string> r = m_daemon->getTransactions(tx_hashes, txs, missed_hashes);
+
+  if (r || txs.size() != 1)
   {
     error = tr("failed to get transaction from daemon");
     return false;
   }
-  cryptonote::blobdata tx_data;
-  bool ok;
-  if (res.txs.size() == 1)
-    ok = epee::string_tools::parse_hexstr_to_binbuff(res.txs.front().as_hex, tx_data);
-  else
-    ok = epee::string_tools::parse_hexstr_to_binbuff(res.txs_as_hex.front(), tx_data);
-  if (!ok)
-  {
-    error = tr("failed to parse transaction from daemon");
-    return false;
-  }
-  crypto::hash tx_hash, tx_prefix_hash;
-  cryptonote::transaction tx;
-  if (!cryptonote::parse_and_validate_tx_from_blob(tx_data, tx, tx_hash, tx_prefix_hash))
-  {
-    error = tr("failed to validate transaction from daemon");
-    return false;
-  }
-  if (tx_hash != txid)
-  {
-    error = tr("failed to get the right transaction from daemon");
-    return false;
-  }
```

Comment:
The transaction hash isn't being calculated here.  To check if each entry in the map {hash: transaction} matches seems a bit redundant.  I could add that in as a check, but at that point there are dozens of other things about it which I could validate as well.  Perhaps at a later time we can add a compile-time flag DO_EXCESSIVE_VALIDATION or something, but for now I think it would just be an unnecessary performance hit indeed.